lib/core/src/chain_swap.rs

```diff
@@ -408,58 +416,132 @@ impl ChainSwapStateHandler {
         Ok(())
     }
 
-    async fn refund(&self, swap: &ChainSwap) -> Result<String, PaymentError> {
-        let amount_sat = swap.receiver_amount_sat;
-        let output_address = self.onchain_wallet.next_unused_address().await?.to_string();
+    pub fn prepare_refund(
+        &self,
+        swap_id: &str,
+        output_address: &str,
+        sat_per_vbyte: u32,
+    ) -> LiquidSdkResult<(u32, u64)> {
+        let swap =
+            self.persister
+                .fetch_chain_swap_by_id(swap_id)?
+                .ok_or(LiquidSdkError::Generic {
+                    err: format!("Swap {} not found", swap_id),
+                })?;
+        match swap.refund_tx_id {
+            Some(refund_tx_id) => Err(LiquidSdkError::Generic {
```

Comment:
We should allow the user refund multiple times. It is useful for example to bump fees.

Reply:
I have a follow up PR #322 which removes this check and monitors the lockup addresses